EcoSoapBank/Helpers/UIKit/UIView+Layout.swift

```diff
@@ -26,4 +31,127 @@ extension UIView {
+    ///
+    /// Also sets `translatesAutoResizingMaskIntoConstraints` to false.
+    func constrainNewSubviewToSides(_ subView: UIView, constant: CGFloat = 0) {
+        constrainNewSubview(
+            subView,
+            with: constraints(for: subView, to: .all, constant: constant))
+    }
+
+    /// Adds provided subview and constrains to anchors equal to provided sides (all by default).
+    ///
+    /// Also sets `translatesAutoResizingMaskIntoConstraints` to false.
+    func constrainNewSubviewToSafeArea(
+        _ subView: UIView,
+        sides: Set<LayoutSide> = .all,
+        constant: CGFloat = 0
+    ) {
+        let constraints = subView.constraints(
+            for: self.safeAreaLayoutGuide,
+            to: sides,
+            constant: constant)
+        constrainNewSubview(subView, with: constraints)
+    }
+}
+
+
+extension UIEdgeInsets {
+    var width: CGFloat { left + right }
+    var height: CGFloat { top + bottom }
+}
+
+
+protocol AutoLayoutConstrainable {
+    var bottomAnchor: NSLayoutYAxisAnchor { get }
+    var centerXAnchor: NSLayoutXAxisAnchor { get }
+    var centerYAnchor: NSLayoutYAxisAnchor { get }
+    var heightAnchor: NSLayoutDimension { get }
+    var leadingAnchor: NSLayoutXAxisAnchor { get }
+    var leftAnchor: NSLayoutXAxisAnchor { get }
+    var rightAnchor: NSLayoutXAxisAnchor { get }
+    var topAnchor: NSLayoutYAxisAnchor { get }
+    var trailingAnchor: NSLayoutXAxisAnchor { get }
+    var widthAnchor: NSLayoutDimension { get }
+}
+
+extension AutoLayoutConstrainable {
+    func constraints(
+        for constrainable: AutoLayoutConstrainable,
+        to sides: Set<LayoutSide> = .all,
+        constant: CGFloat = 0
+    ) -> [NSLayoutConstraint] {
+        sides.map {
+            $0.constraint(from: self, to: constrainable, constant: constant)
+        }
+    }
+}
+
+extension UIView: AutoLayoutConstrainable {}
+extension UILayoutGuide: AutoLayoutConstrainable {}
+
+
+enum LayoutSide: CaseIterable {
+    case top
+    case leading
+    case trailing
+    case bottom
+
+    /// Returns an "equal to" constraint for this side between each of the provided views.
+    func constraint(
+        from constrainable: AutoLayoutConstrainable,
+        to other: AutoLayoutConstrainable,
+        constant: CGFloat = 0
+    ) -> NSLayoutConstraint {
+        switch self {
+        case .top:
+            return constrainable.topAnchor
+                .constraint(equalTo: other.topAnchor, constant: constant)
+        case .leading:
+            return constrainable.leadingAnchor
+                .constraint(equalTo: other.leadingAnchor, constant: constant)
+        case .trailing:
+            return constrainable.trailingAnchor
+                .constraint(equalTo: other.trailingAnchor, constant: -constant)
+        case .bottom:
+            return constrainable.bottomAnchor
+                .constraint(equalTo: other.bottomAnchor, constant: -constant)
+        }
+    }
+}
```

Comment:
I'm leaving this comment on this enum, but you clearly put some work in to designing these helpers. Excited to see them in use to get a better idea of how they all work together. I also think these kind of things take some iteration sometimes after using them for a while to see how you like them, and I will probably pick and choose concepts I like for re-use in future projects.

Reply:
Yeah, this was actually taken from a previous project. I spent maybe 15 minutes yesterday starting a completely new set of types before realizing it would probably take me several days to get to a place I was happy with, so I just decided to keep using these for now.